src/Common/tests/System/Diagnostics/RemoteExecutorTestBase.cs

```diff
@@ -4,17 +4,18 @@
 
 using System.IO;
 using System.Reflection;
+using System.Runtime.InteropServices;
 using Xunit;
 
 namespace System.Diagnostics
 {
     /// <summary>Base class used for all tests that need to spawn a remote process.</summary>
     public abstract class RemoteExecutorTestBase : FileCleanupTestBase
     {
-        /// <summary>The CoreCLR host used to host the test console app.</summary>
-        protected const string HostRunner = "corerun";
         /// <summary>The name of the test console app.</summary>
         protected const string TestConsoleApp = "RemoteExecutorConsoleApp.exe";
+        /// <summary>The CoreCLR host used to host the test console app.</summary>
+        protected static readonly string HostRunner = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "CoreRun.exe" : "corerun";
```

Comment:
I've seen it in a couple places in the targets files for example, https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L37.